cwatch.php

```diff
@@ -417,37 +459,130 @@ public function suspendService($package, $service, $parent_package = null, $pare
      * @see Module::getModule()
      * @see Module::getModuleRow()
      */
-    public function cancelService($package, $service, $parent_package = null, $parent_service = null) 
+    public function cancelService($package, $service, $parent_package = null, $parent_service = null)
     {
         $service_fields = $this->serviceFieldsToObject($service->fields);
-        $licensekey = $service_fields->licensekey;
-        $row = $this->getModuleRow();
-        if (isset($row->meta->username)) {
-            $user = $row->meta->username;
-        }
-        if (isset($row->meta->password)) {
-            $pass = $row->meta->password;
-        }
-        if (isset($package->meta->cwatch_sandbox)) {
-            $sandbox = $package->meta->cwatch_sandbox;
-        }
-        $this->loadApi($user, $pass, $sandbox);
+        // Delete user
+        $this->deleteUser($service_fields->cwatch_email);
```

Comment:
Is this cwatch_email field guaranteed to exist as a service field?

Reply:
Depends on what you mean I guess.  The validation rules require it when adding a service, so yea I'd say so.